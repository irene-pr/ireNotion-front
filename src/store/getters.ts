import router from "@/router";
import paths from "../router/paths";

const getters = {
  redirectToUserBoard(): void {
    router.push(paths.userBoard);
  },
  redirectToLogin(): void {
    router.push(paths.login);
  },
  redirectToRegister(): void {
    router.push(paths.register);
  },
  redirectToHome(): void {
    router.push(paths.home);
  },
};

export default getters;
