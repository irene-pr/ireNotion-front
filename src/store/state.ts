const state = {
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
};

export default state;
