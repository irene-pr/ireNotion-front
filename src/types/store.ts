export interface IChecklist {
  isChecked: boolean;
  paragraph: string;
}

export interface INote {
  type?: string;
  color?: string;
  position?: number;
  title?: string;
  paragraph?: string;
  list?: Array<string | IChecklist>;
  id: string;
  userId?: string;
}

export interface IBoard {
  type: string;
  name: string;
  notes: Array<INote>;
  id: string;
}

export interface IUserContent {
  name: string;
  boards: Array<IBoard>;
}

export interface IUserData {
  userId: string;
  username: string;
}

export interface IState {
  isLoggedIn: boolean;
  userContent: IUserContent;
  userData: IUserData;
  isLoading: boolean;
  themeHeaders: string;
  themeSurfaces: string;
  isBoardEditModal: boolean;
  isUpdateNoteModal: boolean;
  isUpdateListNoteModal: boolean;
  isUpdateChecklistNoteModal: boolean;
  idForModal: string;
  listForModal: Array<string>;
}

export interface IThemeState {
  themeHeaders: string;
  themeSurfaces: string;
}

export interface IUserState {
  isLoggedIn: boolean;
  userContent: IUserContent;
  userData: IUserData;
  isLoading: boolean;
}

export interface IModalState {
  isBoardEditModal: boolean;
  isUpdateNoteModal: boolean;
  isUpdateListNoteModal: boolean;
  isUpdateChecklistNoteModal: boolean;
  idForModal: string;
  listForModal: Array<string>;
}
