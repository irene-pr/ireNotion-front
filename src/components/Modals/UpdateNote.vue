<template>
  <div class="modal" :class="'modal--' + themeSurfaces" @dblclick="closeModal">
    <div
      class="note-update-modal"
      :class="'note-update-modal--' + themeSurfaces"
    >
      <font-awesome-icon
        icon="times"
        class="form__icon-close"
        @click="closeModal"
      ></font-awesome-icon>
      <h2 class="note-update-modal__title">Edit the note</h2>
      <p
        class="note-update-modal__tip"
        :class="'note-update-modal__tip--' + themeSurfaces"
      >
        tip: you can leave empty fields, they won't leave an empty space.
      </p>
      <form
        class="note-update-form-paragraph"
        @submit.prevent="onSubmitParagraph"
        autocomplete="off"
        v-if="this.note.type === 'paragraph'"
      >
        <div class="note-update-form-paragraph__title">
          <label class="note-update-form-paragraph__label" for="title"
            >Title</label
          >
          <input
            class="note-update-form-paragraph__input"
            type="text"
            id="title"
            v-model="note.title"
            placeholder="New title"
          />
        </div>
        <div class="note-update-form-paragraph__text">
          <label class="note-update-form-paragraph__label" for="text"
            >Text
          </label>
          <textarea
            class="note-update-form-paragraph__input"
            type="text"
            id="text"
            v-model="note.paragraph"
            placeholder="New text"
          />
        </div>
        <p>Color</p>
        <div class="note-update-form-paragraph__color">
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="pink"
              >Pink</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="pink"
              value="pink"
              v-model="note.color"
            />
          </div>
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="yellow"
              >Yellow</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="yellow"
              value="yellow"
              v-model="note.color"
            />
          </div>
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="blue"
              >Blue</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="blue"
              value="blue"
              v-model="note.color"
            />
          </div>
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="orange"
              >Orange</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="orange"
              value="orange"
              v-model="note.color"
            />
          </div>
        </div>

        <button
          class="note-update-form-paragraph__button"
          type="submit"
          :class="'note-update-form-paragraph__button--' + themeSurfaces"
          :disabled="note === ''"
        >
          Edit
        </button>
      </form>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions, mapState } from "vuex";
import { IBoard, INote } from "@/types/interfaces";

export default defineComponent({
  name: "UpdateNote",
  data() {
    return {
      note: {
        type: "",
        order: "",
        color: "",
        title: "",
        paragraph: "",
        list: [],
        file: "",
      },
    };
  },
  computed: {
    ...mapState(["userContent", "themeSurfaces", "idForModal"]),
  },
  methods: {
    ...mapActions([
      "updateNote",
      "setUpdateNoteModal",
      "setIdForModal",
      "getUserContent",
    ]),
    onSubmitParagraph() {
      const note = {
        color: this.note.color,
        title: this.note.title,
        paragraph: this.note.paragraph,
      };
      this.updateNote({ idNote: this.idForModal, updatedNote: note });
      this.closeModal();
    },
    closeModal() {
      this.setUpdateNoteModal(false);
      this.setIdForModal("");
    },
  },
  mounted() {
    this.note = {
      ...this.userContent.boards
        .filter(
          ({ notes }: IBoard) =>
            notes.filter(({ id }: INote) => id === this.idForModal).length !== 0
        )[0]
        .notes.filter(({ id }: INote) => id === this.idForModal)[0],
    };
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";
@import "@/styles/_extends";

.modal {
  position: fixed;
  width: 100%;
  height: 100%;
  z-index: 20;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: $Lato;

  &--day-mode {
    background-color: rgba(255, 255, 255, 0.7);

    input {
      color: $theme-dark-color;
    }
    textarea {
      color: $theme-dark-color;
    }
  }
  &--night-mode {
    background-color: rgba(0, 0, 0, 0.7);

    input {
      color: $theme-light-color;
    }
    textarea {
      color: $theme-light-color;
    }
  }
}

.note-update-modal {
  position: relative;
  width: 300px;
  height: 500px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border-radius: 15px;

  &--day-mode {
    background-color: $theme-light-color;
    color: $theme-dark-color;
    border: 2px solid $theme-dark-color-opacity;
  }
  &--night-mode {
    background-color: $theme-dark-color;
    color: $theme-light-color;
    border: 2px solid $theme-light-color-opacity;
  }

  &__title {
    margin-bottom: 0;
    font-weight: 400;
  }
  &__tip {
    padding: 0 40px;
    text-align: center;
    font-size: 13px;
    &--day-mode {
      color: $theme-dark-color-opacity;
    }
    &--night-mode {
      color: $theme-light-color-opacity;
    }
  }
  .note-update-form-paragraph {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    &__title {
      margin-bottom: 20px;
    }
    &__text {
      margin: 0;
      display: flex;
      justify-content: space-between;
      align-items: center;
      textarea {
        resize: none;
        height: 100px;
        width: 180px;
        border: 1px solid $theme-form-line;
        font-family: $Lato;
      }
      label {
        padding-right: 22px;
      }
    }
    p {
      color: $theme-pink;
    }
    &__color {
      margin: 0 0 25px 0;
      display: flex;
      flex-direction: column;

      .radio-container {
        height: 25px;
        display: flex;
        justify-content: space-between;
        align-items: center;
      }
    }
    &__label {
      padding-right: 20px;
    }
    &__input {
      height: 30px;
      padding-bottom: 0;
      border: none;
      border-bottom: 1px solid $theme-form-line;
      background-color: transparent;
      font-size: 14px;

      &:focus {
        outline: none;
        color: $theme-pink;
      }
    }

    &__button {
      @extend %button--little;
      &--day-mode {
        background-color: rgba(255, 255, 255, 0.5);
        color: $theme-dark-color;
        border: 1px solid $theme-dark-color;
      }
      &--night-mode {
        background-color: rgba(0, 0, 0, 0.5);
        color: $theme-light-color;
        border: 1px solid $theme-light-color;
      }
    }
  }

  .form__icon-close {
    position: absolute;
    top: 10px;
    right: 10px;
  }
}
</style>
