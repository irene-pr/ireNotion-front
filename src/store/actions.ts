import { ActionContext } from "vuex";
import axios from "axios";
import jwtDecode from "jwt-decode";
import { IState, IUserLoginData } from "@/types/interfaces";
import router from "@/router";
import paths from "@/router/paths";

const actions = {
  async loginUser(
    { commit }: ActionContext<IState, IState>,
    user: IUserLoginData
  ): Promise<void> {
    const { data: token } = await axios.post(
      `${process.env.VUE_APP_API}/user/login/`,
      user
    );
    localStorage.setItem("token", JSON.stringify(token));
    commit("setUserData", jwtDecode(token.token));
  },

  async getUserContent({
    commit,
  }: ActionContext<IState, IState>): Promise<void> {
    const response = JSON.parse(localStorage.getItem("token") || "");
    const { data: userContent } = await axios.get(
      `${process.env.VUE_APP_API}/user/content/`,
      {
        headers: { Authorization: `Bearer ${response.token}` },
      }
    );
    if (response.status === 401) {
      router.push(paths.login);
    }
    commit("setUserContent", userContent);
  },

  async checkToken({
    commit,
  }: ActionContext<IState, IState>): Promise<string | void> {
    try {
      const token = JSON.parse(localStorage.getItem("token") || "");
      commit("setUserData", jwtDecode(token.token));
      return commit("setIsLoggedIn", true);
    } catch {
      return "User not logged in";
    }
  },

  async createParagraphNote(
    { dispatch }: ActionContext<IState, IState>,
    idBoard: string
  ): Promise<string | void> {
    const randomNumber = Math.floor(Math.random() * 4);
    const colors = ["yellow", "pink", "blue", "orange"];
    const initialParagraphCard = {
      type: "paragraph",
      title: "Title",
      paragraph: "This is a note. You can write whatever you want",
      color: colors[randomNumber],
    };
    const body = {
      note: initialParagraphCard,
      idBoard,
    };

    try {
      const { token } = JSON.parse(localStorage.getItem("token") || "");
      await axios.post(`${process.env.VUE_APP_API}/note/create/`, body, {
        headers: { Authorization: `Bearer ${token}` },
      });
      return dispatch("getUserContent");
    } catch {
      return "Paragraph note could not be created";
    }
  },
};

export default actions;
