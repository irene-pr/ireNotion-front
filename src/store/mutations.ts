import { IState, IUserContent } from "@/types/interfaces";

const mutations = {
  setToken(state: IState, payload: string): void {
    state.token = payload;
  },

  setUserContent(state: IState, payload: IUserContent): void {
    state.userContent = payload;
  },
};

export default mutations;
