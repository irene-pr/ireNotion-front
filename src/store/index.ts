import { createStore } from "vuex";
import actions from "./actions";
import mutations from "./mutations";
import state from "./state";
import getters from "./getters";

export default createStore({
  state,
  mutations,
  actions,
  getters,
  modules: {},
});
