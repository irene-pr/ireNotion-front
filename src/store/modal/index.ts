import state from "./state";
import mutations from "./mutations";
import actions from "./actions";

const modal = {
  namespaced: true,
  state,
  mutations,
  actions,
};

export default modal;
