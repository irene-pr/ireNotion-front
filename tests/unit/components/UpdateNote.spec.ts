import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fas } from "@fortawesome/free-solid-svg-icons";
import UpdateNote from "@/components/Modals/UpdateNote.vue";
import { IState } from "@/types/store";

library.add(fas);

describe("Given a UpdateNote component", () => {
  let mockedState: IState;
  let updateNoteMock = jest.fn();
  let setUpdateNoteModalMock = jest.fn();
  let setIdForModalMock = jest.fn();
  let onSubmitParagraphMock = jest.fn();
  let closeModalMock = jest.fn();
  let store = createStore({});
  let options = {
    data() {
      return {
        type: "",
        order: "",
        color: "",
        title: "",
        paragraph: "",
        list: [],
        file: "",
      };
    },
    components: {
      "font-awesome-icon": FontAwesomeIcon,
    },
    stubs: ["FontAwesomeIcon"],
    global: {
      plugins: [store],
    },
    mocks: {
      $store: store,
    },
  };
  function getOptions() {
    return {
      data() {
        return {
          type: "",
          order: "",
          color: "",
          title: "",
          paragraph: "",
          list: [],
          file: "",
        };
      },
      components: {
        "font-awesome-icon": FontAwesomeIcon,
      },
      stubs: ["FontAwesomeIcon"],
      global: {
        plugins: [store],
      },
      mocks: {
        $store: store,
      },
    };
  }
  function getOptionsWithMethods() {
    return {
      data() {
        return {
          type: "",
          order: "",
          color: "",
          title: "",
          paragraph: "",
          list: [],
          file: "",
        };
      },
      components: {
        "font-awesome-icon": FontAwesomeIcon,
      },
      stubs: ["FontAwesomeIcon"],
      global: {
        plugins: [store],
      },
      methods: {
        onSubmitParagraph: onSubmitParagraphMock,
        closeModal: closeModalMock,
      },
      mocks: {
        $store: store,
      },
    };
  }

  beforeEach(() => {
    mockedState = {
      isLoading: false,
      isLoggedIn: true,
      userContent: {
        name: "Franny",
        boards: [
          {
            type: "board",
            name: "tablero1",
            notes: [
              {
                type: "paragraph",
                color: "yellow",
                title: "title",
                paragraph: "paragraph",
                id: "678",
              },
            ],
            id: "123",
          },
          {
            type: "board",
            name: "tablero2",
            notes: [{ type: "paragraph", color: "yellow", id: "987" }],
            id: "234",
          },
        ],
      },
      userData: {
        userId: "",
        username: "",
      },
      themeHeaders: "night-mode",
      themeSurfaces: "day-mode",
      isBoardEditModal: false,
      isUpdateNoteModal: true,
      idForModal: "678",
    };
    updateNoteMock = jest.fn();
    setUpdateNoteModalMock = jest.fn();
    setIdForModalMock = jest.fn();
    onSubmitParagraphMock = jest.fn();
    closeModalMock = jest.fn();
    store = createStore({
      state() {
        return mockedState;
      },
      actions: {
        updateNote: updateNoteMock,
        setUpdateNoteModal: setUpdateNoteModalMock,
        setIdForModal: setIdForModalMock,
      },
    });
  });

  describe("When it is rendered", () => {
    test("Then it renders", () => {
      options = getOptions();
      mount(UpdateNote, options);
    });
    test("Then the it matches the snapshot", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Then type of note is 'paragraph'", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);
      expect(wrapper.vm.$data.note.type).toBe("paragraph");
    });
    test("Then color of note is 'yellow'", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);
      expect(wrapper.vm.$data.note.color).toBe("yellow");
    });
    test("Then title of note is 'title'", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);
      expect(wrapper.vm.$data.note.title).toBe("title");
    });
    test("Then paragraph of note is 'paragraph'", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);
      expect(wrapper.vm.$data.note.paragraph).toBe("paragraph");
    });
    test("Then it renders a closing icon", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const closeIcon = wrapper.find(".form__icon-close");

      expect(closeIcon.exists()).toBeTruthy();
    });
    test("Then it renders a heading", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const heading = wrapper.find("h2");

      expect(heading.text()).toBe("Edit the note");
    });
    test("Then it renders a tp text", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const tip = wrapper.find("p");

      expect(tip.text()).toBe(
        "tip: you can leave empty fields, they won't leave an empty space."
      );
    });

    test("Then it renders a form", async () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const form = await wrapper.find("form");

      expect(form.exists()).toBeTruthy();
    });
  });
  describe("When the form is rendered", () => {
    test("Then a label for title is rendered", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const label = wrapper.find("label[for='title']");

      expect(label.text()).toBe("Title");
    });
    test("Then an input for title is rendered", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const input = wrapper.find("input[id='title']");

      expect(input.exists()).toBeTruthy();
    });
    test("Then a label for text is rendered", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const label = wrapper.find("label[for='text']");

      expect(label.text()).toBe("Text");
    });
    test("Then a textarea for text is rendered", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const textarea = wrapper.find("textarea[id='text']");

      expect(textarea.exists()).toBeTruthy();
    });
    test("Then 5 inputs radio are rendered", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const radio = wrapper.findAll("input[type='radio']");

      expect(radio).toHaveLength(5);
    });
    test("Then 5 labels for colors are rendered", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const labelPink = wrapper.find("label[for='pink']");
      const labelYellow = wrapper.find("label[for='yellow']");
      const labelBlue = wrapper.find("label[for='blue']");
      const labelOrange = wrapper.find("label[for='orange']");
      const labelGreen = wrapper.find("label[for='green']");

      expect(labelPink.exists()).toBeTruthy();
      expect(labelBlue.exists()).toBeTruthy();
      expect(labelGreen.exists()).toBeTruthy();
      expect(labelOrange.exists()).toBeTruthy();
      expect(labelYellow.exists()).toBeTruthy();
    });
    test("Then an edit button is rendered", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const button = wrapper.find("button");

      expect(button.exists()).toBeTruthy();
    });
  });
  describe("When it renders a closing icon and you click it", () => {
    test("Then the method closeModal is called", () => {
      options = getOptionsWithMethods();
      const wrapper = mount(UpdateNote, options);

      const closeIcon = wrapper.find(".form__icon-close");
      closeIcon.trigger("click");

      expect(closeModalMock).toHaveBeenCalled();
    });
    test("Then the action setUpdateNoteModal is called", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const closeIcon = wrapper.find(".form__icon-close");
      closeIcon.trigger("click");

      expect(setUpdateNoteModalMock).toHaveBeenCalled();
    });
    test("Then the action setUpdateNoteModal is called", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const closeIcon = wrapper.find(".form__icon-close");
      closeIcon.trigger("click");

      expect(setIdForModalMock).toHaveBeenCalled();
    });
  });
  describe("When the modal is double clicked", () => {
    test("Then the method closeModal is called", () => {
      options = getOptionsWithMethods();
      const wrapper = mount(UpdateNote, options);

      const modal = wrapper.find(".modal");
      modal.trigger("dblclick");

      expect(closeModalMock).toHaveBeenCalled();
    });
    test("Then the action setUpdateNoteModal is called", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const modal = wrapper.find(".modal");
      modal.trigger("dblclick");

      expect(setUpdateNoteModalMock).toHaveBeenCalled();
    });
    test("Then the action setIdForModal is called", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const modal = wrapper.find(".modal");
      modal.trigger("dblclick");

      expect(setIdForModalMock).toHaveBeenCalled();
    });
  });

  describe("When input has been rewritten and the edit button clicked", () => {
    test("Then the onSubmit method gets called", () => {
      options = getOptionsWithMethods();
      const wrapper = mount(UpdateNote, options);

      const input = wrapper.find("input");
      input.setValue("Mario");

      const form = wrapper.find("form");
      form.trigger("submit.prevent");

      expect(onSubmitParagraphMock).toHaveBeenCalled();
    });
    test("Then the onSubmit method gets called", () => {
      options = getOptions();
      const wrapper = mount(UpdateNote, options);

      const input = wrapper.find("input");
      input.setValue("Mario");

      const form = wrapper.find("form");
      form.trigger("submit.prevent");

      expect(updateNoteMock).toHaveBeenCalled();
    });
  });
});
