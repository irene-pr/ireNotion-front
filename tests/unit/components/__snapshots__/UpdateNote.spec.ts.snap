// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a UpdateNote component When it is rendered Then the it matches the snapshot 1`] = `
<div
  class="modal modal--day-mode"
>
  <div
    class="note-update-modal note-update-modal--day-mode"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 form__icon-close"
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class=""
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
    <h2
      class="note-update-modal__title"
    >
      Edit the note
    </h2>
    <p
      class="note-update-modal__tip note-update-modal__tip--day-mode"
    >
       tip: you can leave empty fields, they won't leave an empty space. 
    </p>
    <form
      autocomplete="off"
      class="note-update-form-paragraph"
    >
      <div
        class="note-update-form-paragraph__title"
      >
        <label
          class="note-update-form-paragraph__label"
          for="title"
        >
          Title
        </label>
        <input
          class="note-update-form-paragraph__input"
          id="title"
          placeholder="New title"
          type="text"
        />
      </div>
      <div
        class="note-update-form-paragraph__text"
      >
        <label
          class="note-update-form-paragraph__label"
          for="text"
        >
          Text 
        </label>
        <textarea
          class="note-update-form-paragraph__input"
          id="text"
          placeholder="New text"
          type="text"
        />
      </div>
      <p>
        Color
      </p>
      <div
        class="note-update-form-paragraph__color"
      >
        <div
          class="radio-container"
        >
          <label
            class="note-update-form-paragraph__label"
            for="pink"
          >
            Pink
          </label>
          <input
            class="note-update-form-paragraph__input"
            id="pink"
            type="radio"
            value="pink"
          />
        </div>
        <div
          class="radio-container"
        >
          <label
            class="note-update-form-paragraph__label"
            for="yellow"
          >
            Yellow
          </label>
          <input
            class="note-update-form-paragraph__input"
            id="yellow"
            type="radio"
            value="yellow"
          />
        </div>
        <div
          class="radio-container"
        >
          <label
            class="note-update-form-paragraph__label"
            for="blue"
          >
            Blue
          </label>
          <input
            class="note-update-form-paragraph__input"
            id="blue"
            type="radio"
            value="blue"
          />
        </div>
        <div
          class="radio-container"
        >
          <label
            class="note-update-form-paragraph__label"
            for="orange"
          >
            Orange
          </label>
          <input
            class="note-update-form-paragraph__input"
            id="orange"
            type="radio"
            value="orange"
          />
        </div>
        <div
          class="radio-container"
        >
          <label
            class="note-update-form-paragraph__label"
            for="green"
          >
            Green
          </label>
          <input
            class="note-update-form-paragraph__input"
            id="green"
            type="radio"
            value="green"
          />
        </div>
      </div>
      <button
        class="note-update-form-paragraph__button note-update-form-paragraph__button--day-mode"
        type="submit"
      >
         Edit 
      </button>
    </form>
  </div>
</div>
`;
